components/calendar/calendar-header.spec.ts

```diff
@@ -212,9 +257,22 @@ class NzTestCalendarHeaderActiveDateComponent {
 }
 
 @Component({
-  template: ` <nz-calendar-header (yearChange)="year = $event" (monthChange)="month = $event"></nz-calendar-header> `
+  template: `
+    <nz-calendar-header
+      [nzCustomHeader]="customHeader"
+      (yearChange)="year = $event"
+      (monthChange)="month = $event"
+    ></nz-calendar-header>
+
+    <ng-template #custom>
+      <p>custom</p>
+    </ng-template>
```

Comment:
```suggestion
    <nz-calendar-header
      [nzCustomHeader]="customHeader"
      (yearChange)="year = $event"
      (monthChange)="month = $event"
    ></nz-calendar-header>

    <ng-template #customHeader>
      <p>custom</p>
    </ng-template>
```

Reply:
that `ng-template` should not have the same name as `customHeader` because `customHeader` is already declared in [here](https://github.com/NG-ZORRO/ng-zorro-antd/pull/8418/files#diff-fbfd5702bfd320dfafa4c694dcf37f2ab6693eafb81e8ff234176dcf94cd80cdR263).

The reason as to why I have declared two diffrent set of custom header is because of this [line](https://github.com/NG-ZORRO/ng-zorro-antd/pull/8418/files#diff-fbfd5702bfd320dfafa4c694dcf37f2ab6693eafb81e8ff234176dcf94cd80cdR205).
Although I changed the name to avoid confusion.